File: Back-end/biblioteca/src/main/java/com/managerServices/biblioteca/exception/HttpReturnException.java
package com.managerServices.biblioteca.exception;

import org.springframework.http.HttpStatus;

public class HttpReturnException extends Exception {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	
	
	public static final int OK = 200;
	public static final int CONFLICT = 409;
	public static final int NOT_FOUND = 404;

	private int status_code;
	private HttpStatus status;
	private String message;
	
	public int getStatus_code() {
		return status_code;
	}
	public void setStatus_code(int status_code) {
		this.status_code = status_code;
	}
	public HttpStatus getStatus() {
		return status;
	}
	public void setStatus(HttpStatus status) {
		this.status = status;
	}
	public String getMessage() {
		return message;
	}
	public void setMessage(String message) {
		this.message = message;
	}
	public HttpReturnException() {
		super();
	}
	public HttpReturnException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
		super(message, cause, enableSuppression, writableStackTrace);
	}
	public HttpReturnException(String message, Throwable cause) {
		super(message, cause);
	}
	public HttpReturnException(String message) {
		super(message);
	}
	public HttpReturnException(Throwable cause) {
		super(cause);
	}
	public HttpReturnException(int status_code, HttpStatus status, String message) {
		super();
		this.status_code = status_code;
		this.status = status;
		this.message = message;
	}	
	
}

File: Back-end/biblioteca_kotlin/library/src/main/kotlin/com/biblioteca/library/exception/HttpReturnException.kt
package com.biblioteca.library.exception

import org.springframework.http.HttpStatus
import java.lang.Exception

class HttpReturnException(status_code: Int, status: HttpStatus, message: String) : Exception() {
	
	

	override fun hashCode(): Int {
		return super.hashCode()
	}

	var status_code = status_code
	var status = status
	override var message = message
	
	companion object {
		final val OK = 200
		final val CONFLICT = 409
		final val NOT_FOUND = 404;
	 	final val NO_CONTENT = 204;
	 	final val INTERNAL_SERVER_ERROR = 500;

	 	final val SQL_ERR_01 = "Não possui nenhum aluguel de livro";
	
	}
	
}